### ISSUE TITLE
Server stalls if file name for cache.disk() is too long

### ISSUE THREAD

**Issue by sfxxx** [web2py/web2py#1393]
web2py 2.14.6:
In the interactive debugger, I get an exception, but no ticket:

Interaction at /Users/sfx/dev/mdb/web2py.app/Contents/Resources/gluon/cache.py line 473

Exception IOError: (63, 'File name too long', "/Users/sfx/dev/mdb/web2py.app/Contents/Resources/applications/mdb/cache/05e/1cc/(Score([(u'Hosho', [u'', u'.', u'', u'', u'.', u'', u'', u'.', u'', u'', u'.', u'', u'', u'.', u'', u'', u'.', u'', u'', u'.', u'', u'', u'.', u'', u'', u'.', u'', u'', u'.', u'', u'', u'.', u'', u'', u'.', u'', u'', u'.', u'', u'', u'.', u'', u'', u'.', u'', u'', u'.', u'']), (u'RI', [u'3', u'', u'', u'5', u'', u'', u'3', u'', u'', u'5', u'', u'', u'3', u'', u'', u'5', u'', u'', u'3', u'', u'', u'6', u'', u'', u'3', u'', u'', u'5', u'', u'', u'5', u'', u'', u'', u'3', u'', u'3', u'', u'3', u'', u'', u'', u'', u'', u'', u'', u'3', u'']), (u'RT', [u'3', u'1', u'', u'', u'1', u'', u'3', u'', u'', u'', u'2', u'', u'3', u'1', u'', u'', u'1', u'', u'3', u'', u'', u'', u'1', u'', u'3', u'1', u'', u'', u'', u'', u'', u'', u'', u'', u'3', u'', u'3', u'', u'3', u'', u'2', u'2', u'', u'', u'1', u'', u'3', u'']), (u'L', [u'', u'1', u'', u'5', u'', u'', u'', u'7', u'', u'5', u'', u'', u'', u'1', u'', u'5', u'', u'', u'', u'7', u'', u'6', u'', u'', u'', u'1', u'', u'5', u'', u'', u'4', u'', u'', u'6', u'', u'', u'6', u'', u'', u'6', u'', u'', u'6', u'1', u'', u'6', u'', u'']), (u'B', [u'', u'', u'', u'', u'', u'', u'', u'', u'', u'', u'5', u'', u'', u'', u'', u'', u'', u'', u'', u'', u'', u'', u'6', u'', u'', u'', u'', u'', u'5', u'', u'', u'4', u'', u'', u'6', u'', u'', u'6', u'', u'', u'2', u'', u'', u'', u'', u'', u'6', u''])]), 2, 4, 240, 'ogg')")

**Comment by niphlod** [comment]
I wonder if it's time for a cache that doesn't accept as a key anything but a string shorter than 200 chars ?!

**Comment by mdipierro** [comment]
This depends on the filesystem. If you know you FS has limitations, please hash the key.

**Comment by sfxxx** [comment]
Yes, that's what I did. However, the problem occurred after upgrading my production server from 2.9.5 to 2.14.6, and it took a bit of debugging to figure out. In 2.9.5 the same long keys didn't seem to be a problem.

The fact that the server stalls without issuing a ticket is not ideal, don't you think so? Clearly there is some exception raised (see above), but it seems to be not caught properly. I didn't look deeper, but I thought this may point to a more general problem. That's why I reported this an issue.
